app/src/main/java/org/fossasia/openevent/general/auth/AuthService.kt

```diff
@@ -33,21 +34,37 @@ class AuthService(private val authApi: AuthApi,
     fun isLoggedIn() = authHolder.isLoggedIn()
 
     fun logout() {
-        userMap.remove(authHolder.getId())
-
+        val userFlowable = userDao.getUser(authHolder.getId())
+        userFlowable.map {
+            Single.fromCallable {
+                userDao.deleteUser(it)
+            }.subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe({
+                        Timber.d("User logged out!")
+                    }
+                    ) {
+                        Timber.e(it, "Error logging out user!")
+                    }
+        }
```

Comment:
Logic is not correct, also, return the completable from here and subscribe on the caller site

Reply:
okay